data/account/participationRegistry.go

```diff
@@ -524,8 +561,19 @@ func (db *participationDB) Insert(record Participation) (id ParticipationID, err
 		insert:   record,
 	}
 
+	// Make some copies.
+	var vrf crypto.VRFSecrets
+	if record.VRF != nil {
+		copy(vrf.SK[:], record.VRF.SK[:])
+		copy(vrf.PK[:], record.VRF.PK[:])
+	}
+
+	var voting crypto.OneTimeSignatureSecrets
+	if record.Voting != nil {
+		voting = record.Voting.Snapshot()
+	}
+
 	// update cache.
-	// TODO: simplify to re-initializing with initializeCache()?
 	db.cache[id] = ParticipationRecord{
```

Comment:
this looks an awful lot like .Duplicate() ?
(and also things should only be assigned if the source was not nil)

Reply:
Yes, I didn't see a good way to reuse it.